### ISSUE TITLE
Start-AzureRmVM and Stop-AzureRmVM execution requires waiting until machine is up/down

### ISSUE THREAD

**Issue by bergmeister** [Azure/azure-powershell#4600]
### Cmdlet(s)
 - Start-AzureRmVM
 - Stop-AzureRmVM
### Module Version
Current (4.3.1)
### Description
Currently the`Start/Stop-AzureRmVM` cmdlets are blocking until the VM is up/down and although one can put their execution in a `PSRemotingJob`, one still has to wait until the cmdlet has finished. If one closes e.g. the `PowerShell` window before the background job has finished (and therefore the job dies), the Azure VM does not start/stop. I am not interested in knowing when the VM is up/down but only want to send a start/stop signal. The use case is starting/stopping VM using an [Azure PowerShell function](https://azure.microsoft.com/en-gb/services/functions/), where I would have to pay for all the time when it is waiting. In Azure I can simply press the Start/Stop button and close the browser so it should technically be possible to do that.

**Comment by cormacpayne** [comment]
@bergmeister as of AzureRM version 4.4.0, we now support using jobs to execute cmdlets in parallel and in the background of a PowerShell session. Please update to the latest version of AzureRM. For more information about using jobs, please see [here](https://docs.microsoft.com/en-us/powershell/azure/context-persistence#running-azure-powershell-cmdlets-as-background-jobs).

Please feel free to re-open this issue if you feel this is not the functionality you are looking for.

**Comment by bergmeister** [comment]
@cormacpayne Please read the description again. I said explicitly that running it as a background job does not work if I then close the PowerShell session before the command has finished. Every second that it has to unnecessarily wait is costly when using Azure PowerShell functions

**Comment by boqsernmcpopsern** [comment]
@bergmeister late to the party, and you probably know now. You can use -AsJob to skip the wait.

**Comment by bergmeister** [comment]
Yes, thanks, I read that in another issue.

**Comment by mklement0** [comment]
As an aside: `Start-AzureVM` is a misnomer in that PowerShell's nomenclature calls for the `Start` verb to exhibit _asynchronous_ behavior:

From https://docs.microsoft.com/en-us/powershell/developer/cmdlet/approved-verbs-for-windows-powershell-commands (emphasis added):

> **Invoke vs. Start**
The `Invoke` verb is used to perform an operation that is generally a synchronous operation, such as running a command. The **`Start` verb is used to begin an operation that is generally an asynchronous operation**, such as starting a process.

Note that the core `Start-Sleep` cmdlet is similarly misnamed.

---

With synchronous-by-default behavior, there is no good naming, solution, because `Invoke-AzureVm` would be confusing.

The better approach - which would obviously be a breaking change - would have been to make the cmdlet _asynchronous by default_ and offer a `-Wait` switch for synchronous operation.
